src/boot/efi/util.h

```diff
@@ -24,6 +24,17 @@
 #define UINT64_MAX ((UINT64) -1)
 #endif
 
+/* gnu-efi has no format specifier for (U)INTN. */
+#if __SIZEOF_POINTER__ == 4
+#  define FMT_U L"%u"
+#  define FMT_D L"%d"
+#elif __SIZEOF_POINTER__ == 8
+#  define FMT_U L"%lu"
+#  define FMT_D L"%ld"
+#else
+#  error "Unexpected pointer size"
+#endif
```

Comment:
hmm, so C99 PRIu32-style macros do not include the `%` prefix. Our FMT_UID, … macros generally do. That has a reason: the PRIu32-style macros are generic, and still allow you to specify alignment, precision and so on. THe FMT_UID macros on the other hand don't allow that, because we now pretty excactly how UIDs should be formatted.

Now, this here is a generic use case, it is a way to display any natively sized integer. Hence it probably should not contain the "%" prefix.

Hence maybe do:

```c
#define PRIuN L"u"
```

and similar instead, and then use it as:

```c
SPrint(…, "foo %" PRIuN " bar");
```

pretty? no! but I think the most obvious way to do this for people who have accepted C99 syle PRI* macros into their lives.